src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2843,6 +2847,27 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         e.DragUIOverride().IsGlyphVisible(false);
     }
 
+    // Method description:
+    // - Checks if the uri is valid and sends an event if so
+    // Arguments:
+    // - The uri
+    void TermControl::_HyperlinkHandler(std::wstring uri)
+    {
+        try
+        {
+            auto parsed = winrt::Windows::Foundation::Uri(uri);
+            if (parsed.SchemeName() == L"http" || parsed.SchemeName() == L"https")

```

Comment:
Might want to add `file://` as a valid scheme as I can see a lot of scripts giving a hyperlink to a log/results file that they generated on the local machine as part of the build.

Might also want to consider making the valid schemes configurable as someone might want to add other ones for whatever reason (`skype://`, `steam://`, etc.)

Reply:
the `file://` URI is a bit more complicated actually:
https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda#file-uris-and-the-hostname

We'd _need_ to enforce the same hostname policy (I'm thinking that this will be important for #3158 as well)

Reply:
Ugh gross. OK, maybe don't put it in by default and let people override it for now and file a follow-on to make `file://` smarter.